CoroutineTemplate/domain/src/main/java/co/nimblehq/coroutine/domain/usecase/GetUsersUseCase.kt

```diff
@@ -1,24 +0,0 @@
-package co.nimblehq.coroutine.domain.usecase
-
-import co.nimblehq.coroutine.domain.data.entity.UserEntity
-import co.nimblehq.coroutine.domain.dispatcher.IoDispatcher
-import co.nimblehq.coroutine.domain.repository.UserRepository
-import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.withContext
-import javax.inject.Inject
-
-class GetUsersUseCase @Inject constructor(
-    private val userRepository: UserRepository,
-    @IoDispatcher private val ioDispatcher: CoroutineDispatcher
-) {
-    suspend fun execute(): UseCaseResult<List<UserEntity>> {
-        return withContext(ioDispatcher) {
-            try {
-                val result = userRepository.getUsers()
-                UseCaseResult.Success(result)
-            } catch (exception: Exception) {
-                UseCaseResult.Error(exception)
```

Comment:
check this intention again, @doannimble - you are changing the return type and the rule we intended to set on the responsibility of the [Repository - UseCase]. I think it is out of the scope of the PR again

Reply:
@sleepylee Agree, it should be in another PR with further discussion.
I will keep the previous approach to avoid making more complicated.